dialer.go

```diff
@@ -193,6 +193,7 @@ func (d *Dialer) LookupPartition(ctx context.Context, network string, address st
 					return
 				}
 			}
+			break
 		}
 
 		errch <- UnknownTopicOrPartition
```

Comment:
Could we remove this line instead of adding a `break` ? That way we don't create new conditions in the code that have never been exercised?